http/src/util/service/handler.rs

```diff
@@ -106,6 +106,8 @@ where
     T: FromRequest<'static, Req, Error = Err>,
     F: AsyncFn<T::Type<'a>, Output = Res> + Clone,
     F: AsyncFn<T>, // second bound to assist type inference to pinpoint T
+    F: 'a,
+    Req: 'a,
```

Comment:
are theses bounds  necessary?

HRTBs with nested lifetimes breaks badly with any lifetime constraints on impls. I faced a simliar problem recently <https://rust-lang.zulipchat.com/#narrow/stream/131828-t-compiler/topic/tracking.20hrtb.20known.20limitations> .

Reply:
yes. with some new nightly changes this is mandatory now. matser branch now fails to comiple with
```rust
cargo build --package xitca-http --features util-service
```